nss-day-cohort-46/rare-server-tragic-trolls
# Nc inital posts unapproved unless admin

# Description
New posts are not automatically approved when created. Unless you are an admin.
## Type of change

# Testing Instructions

Ensure you have two users in your database. One as an admin and one as an author. Here are two:

```
INSERT INTO Users
  ('first_name', 'last_name', 'email', 'bio', 'username', 'password', 'created_on', 'active', 'is_admin')
VALUES  
  ('test', 'user', 'test@user.com', 'bio', 'test@user.com', 'password', '12/31/2020', True, True);
INSERT INTO Users
  ('first_name', 'last_name', 'email', 'bio', 'username', 'password', 'created_on', 'active', 'is_admin')
VALUES  
  ('test', 'user', 'test@user.com', 'bio', 'test@user.com', 'password', '12/31/2020', True, False);
```

Perform a POST in Postman. Do it once as an admin:
```
http://localhost:8088/posts
```
```
{
        "title": "First attempt as author",
        "imageUrl": "",
        "tagIds": [2],
        "userId": USE-YOUR-ADMIN-USER-ID ,
        "categoryId": 1,
        "content": "Post Text",
        "publicationDate": "2026-04-02T23:57:07.374Z"
}
```
see that the returned object has an approved status of "true"
Do it again as an author
```
{
        "title": "First attempt as author",
        "imageUrl": "",
        "tagIds": [2],
        "userId": USE-YOUR-AUTHOR-USER-ID ,
        "categoryId": 1,
        "content": "Post Text",
        "publicationDate": "2026-04-02T23:57:07.374Z"
}
```
See that the returned result says approval is false.

# Checklist:
